app/src/main/java/com/github/sdpteam15/polyevents/database/observe/Observable.kt

```diff
@@ -1,24 +1,40 @@
 package com.github.sdpteam15.polyevents.database.observe
 
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
 import androidx.lifecycle.LifecycleOwner
-import com.github.sdpteam15.polyevents.helper.HelperFunctions.observeOnDestroy
+import androidx.lifecycle.OnLifecycleEvent
 import com.github.sdpteam15.polyevents.helper.HelperFunctions.run
 
 /**
  * Observable live data of type T
  * @param
  */
 class Observable<T>(value: T? = null, sender: Any? = null) {
-    constructor(value: T? = null) : this(value, null)
-    constructor() : this(null)
+    class AfterRemovable<U>(val then: U, val remove: () -> Boolean)
+    class UpdateValue<T>(val value: T, val sender: Any?)
```

Comment:
/**
 * Data that notify a set of observers on a modification
 * When adding observers and passing a lifecycleOwner in the parameters, the added observer will not be notified if the lifecycle is destroyed.
 * This is useful for example when an activity is stopped, as we don't need to update data on this closed activity.
 * If we don't give a Lifecycle in parameters, the observer will keep beeing notified even if the activity is stopped.
 * To unsubscribe from the Observable, we take take the reference of the corresponding "remove" function
 * We get the "remove" function by taking the return value of the observe method, for example :
 *
 *      val ret : ThenOrRemove = observable.observe { ... }
 *      ...
 *      do something
 *      ...
 *      //on a special event stop observing
 *      ret.remove()
 *
 *  The chain operator "then" from this return function can be used to add multiple observers in a convenient way
 *  For example in an activity:
 *
 *      observable.map(this){ ... }.then.observe(this){ ... }.then.map(this){}.then.observeOnce(this){ ... }
 *
 * When setting a value, we use an "UpdateValue" object which is a pair containing the new value,
 * and eventually a reference to the sender object, which can be used when we need to know who sent the update.
 *
 * @param value initial value of the data
 * @param sender last object that modified the data
 */
class Observable<T>(value: T? = null, sender: Any? = null) {

    /**
     * The return value of each function of an observable
     *
     * @param then returns a reference of the observable after calling a transformation
     * @param remove function to remove the observer from the list
     */
    class AfterRemovable<U>(val then: U, val remove: () -> Boolean)

    /**
     * We use an UpdateValue object each time we want to set a new value for the data.
     * It contains the new value and eventually the object that set the new value.
     *
     * @param value new value of the data
     * @param sender function to remove the observer from the list
     */
    class UpdateValue<T>(val value: T, val sender: Any?)